test/cpp/end2end/client_callback_end2end_test.cc

```diff
@@ -176,6 +183,34 @@ class ClientCallbackEnd2endTest
     }
   }
 
+  void CallResponseCallback() {
+    gpr_log(GPR_ERROR, "KRS: CallResponseCallback");
+    std::lock_guard<std::mutex> l1(mu1);
+    ResponseCallback();
```

Comment:
This is a deadlock caused by the function call, not by the callback. Because this calls ReponseCallback while locked and then ResponseCallback tries to take the lock.

Reply:
Yeah, I think the previous version did not capture my intent. I have modified it now. 
ReponseCallback is being called from the test now which makes the RPC call and we still see the deadlock.

Reply:
Still has the same issue because the callback calls CallResponseCallback which takes the lock and then calls ResponseCallback which again takes the same lock.